### ISSUE TITLE
False positive: "Does not support item assignment" error with empty_like from numpy

### ISSUE THREAD

**Issue by bitsandscraps** [PyCQA/pylint#5543]
### Bug description

reproduce.py
```python
# pylint: disable=missing-docstring
from numpy import arange, empty_like

a = empty_like(arange(10))
a[0] = 1
```

### Command used

```shell
pylint reproduce.py
```

### Pylint output

```shell
************* Module reproduce
reproduce.py:5:0: E1137: 'a' does not support item assignment (unsupported-assignment-operation)
```

### Expected behavior

`python reproduce.py` runs without errors
Also, if we use `import numpy as np` and `np.empty_like` instead of `from numpy import empty_like`, pylint shows no errors.

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.0
Python 3.9.5 (default, Nov 18 2021, 16:00:48) 
[GCC 10.3.0]
```

### OS / Environment

Ubuntu 21.04

### Additional dependencies

astroid==2.9.0
isort==5.10.1
lazy-object-proxy==1.7.1
mccabe==0.6.1
numpy==1.21.4
platformdirs==2.4.0
pylint==2.12.2
toml==0.10.2
typing-extensions==4.0.1
wrapt==1.13.3

**Comment by Pierre-Sassoulas** [comment]
The difference in behavior when importing another way make it looks like we needed to infer at some point and we're not.
